tests/acceptance/features/bootstrap/WebDav.php

```diff
@@ -4651,6 +4652,149 @@ public function downloadPreviewOfFiles(string $user, string $path, string $width
 		$this->setResponse($response);
 	}
 
+	/**
+	 * @When user :user downloads the preview of shared resource :path with width :width and height :height using the WebDAV API
+	 *
+	 * @param string $user
+	 * @param string $path
+	 * @param string $width
+	 * @param string $height
+	 *
+	 * @return void
+	 */
+	public function userDownloadsThePreviewOfSharedResourceWithWidthAndHeightUsingTheWebdavApi(string $user, string $path, string $width, string $height): void {
+		if ($this->getDavPathVersion() === 3) {
+			$this->setResponse($this->userDownloadsOrUploadsSharedResource($user, $path, 'GET', $width, $height));
+		} else {
+			$this->setResponse($this->downloadPreviews($user, $path, null, $width, $height));
+		}
+	}
+
+	/**
+	 * @Given user :user has downloaded the preview of shared resource :path with width :width and height :height
+	 *
+	 * @param string $user
+	 * @param string $path
+	 * @param string $width
+	 * @param string $height
+	 *
+	 * @return void
+	 */
+	public function userHasDownloadedThePreviewOfSharedResourceWithWidthAndHeight(string $user, string $path, string $width, string $height) {
+		$this->userDownloadsThePreviewOfSharedResourceWithWidthAndHeightUsingTheWebdavApi($user, $path, $width, $height);
+		$this->theHTTPStatusCodeShouldBe(200);
```

Comment:
can we make the helper function that returns the response and use the response to set or check according to the step? As i am currently working on the refactoring, it would be great to do that while adding new steps like this.
CC: @saw-jan @SwikritiT  Thoughts on this??

Reply:
Would be great :+1: